pkg/kubelet/prober/prober.go

```diff
@@ -98,24 +98,34 @@ func (pb *prober) probe(ctx context.Context, probeType probeType, pod *v1.Pod, s
 	}
```

Comment:
Code on the lines 95-98 looks unreachable to me as probeSpec can't be nil at this point. I'd suggest to remove it.

Reply:
My bad. `probeSpec` can still be nil if there if probe is not configured for the container. Please, revert this change. This should fix nil pointer dereference you're debugging.